Monika-After-Story/MonikaModDev
# Surprises and imsorry to characters folder + optimizations

## Changes:
- Moves all surprise.txt files and the imsorry file bits to the characters folder update script will move them if they're in the DDLC directory to keep legit `imsorry`s and such)
- Added MASPoems for each individual surprise.txt
- MASPoems have two new properties, `ex_props` and `paper`
  - Poem `ex_props` are used to path dialogue for the `monika_showpoem` topic
  - Paper allows custom paper specific to a single poem and will override the default in the paper category map
- Fixed the sorting of poems in the menu where you select the poem you want to read again
- Moved `is_file_present` to `mas_utils` as was TODO'd
- Cleaned up base some DDLC functions which we have and added documentation for them
- Optimized the `mas_surprise()` function such that it uses a more switch/case format rather than an `if`/`elif`/`elif`/`...`/`else` chain
- Poems for surprises are currently initialized, but unused. I have added a TODO in the unlock function for them.

## Testing:
- Verify that calling the update script transfers all surprise/imsorry files correctly ~~and unlocks the correct poems for you (This is based upon which surprise txts exist in the characters folder (when calling the update script, that is)~~
- ~~Verify that the surprise topic will unlock the respective poem for you after getting it from there~~
- Verify that surprises are exported to the characters folder
- Verify that imsorry is correctly enforced by the `imsorry.txt` being in the characters folder
- ~~Verify that the surprise poem for the Broken affection and Love affection levels both use different paper from the normal~~
- Verify that the below normal affection surprise poems have different lead-in/out dialogue when asking to see it again (you'll have to manually unlock one of the poems for this)
- Verify no crashes
- *And because we did it so well last time...* Verify that all cases for uses of `is_file_present` are still working due to the storemodule change
